custom_components/tuya_local/devices/petsnowy_snowplus_catlitter.yaml

```diff
@@ -0,0 +1,214 @@
+          - dps_val: 2
+            value: Automatic cleaning completed
+          - dps_val: 4
+            value: Timed cleaning completed
+          - dps_val: 8
+            value: Manual cleaning canceled
+          - dps_val: 16
+            value: Manual deodorization completed
+          - dps_val: 32
+            value: Automatic deodorization completed
+          - dps_val: 64
+            value: Timed deodorization completed
+          - dps_val: 128
+            value: Emptying cat litter completed
+          - dps_val: 256
+            value: Cat litter weight reset
+          - dps_val: 512
+            value: Self-check normal
+          - dps_val: 1024
+            value: Button locked
+          - dps_val: 2048
+            value: Button unlocked
+          - dps_val: 4096
+            value: Indicator light on
+          - dps_val: 8192
+            value: Indicator light off
+          - dps_val: 16384
+            value: Pause cleaning
+          - dps_val: 32768
+            value: Sleep mode on
+          - dps_val: 65536
+            value: Sleep mode off
+          - dps_val: 131072
+            value: Cancel emptying cat litter
+          - dps_val: 262144
+            value: Change cat litter
+          - value: Unknown
+  - entity: binary_sensor
+    class: problem
+    category: diagnostic
+    dps:
+      - id: 22
+        name: sensor
+        type: bitfield
+        mapping:
+          - dps_val: 0
+            value: false
+          - value: true
+      - id: 22
+        name: description
+        type: bitfield
+        mapping:
+          - dps_val: 0
+            value: OK
+          - dps_val: 1
+            value: Top cover fault
+          - dps_val: 2
+            value: Drawer fault
+          - dps_val: 4
+            value: Drawer full
+          - dps_val: 8
+            value: Stuck cat
+          - dps_val: 16
+            value: Checking fault
+          - dps_val: 32
+            value: Stay long
+          - dps_val: 64
+            value: Trouble removal
+          - value: Unknown
+  - entity: sensor
+    name: Status
+    class: enum
+    category: diagnostic
+    dps:
+      - id: 24
+        name: sensor
+        type: string
+  - entity: button
+    name: Filter reset
+    icon: "mdi:filter-cog"
+    dps:
+      - id: 101
+        type: boolean
+        name: button
+  - entity: sensor
+    name: Filter life
+    category: diagnostic
+    class: duration
+    dps:
+      - id: 102
+        name: sensor
+        type: integer
+        unit: day
+        range:
+          min: 0
+          max: 60
+  - entity: switch
+    name: Unlock physical buttons
+    icon: "mdi:lock-question"
+    category: config
+    dps:
+      - id: 104
+        type: boolean
+        name: switch
+  - entity: switch
+    name: Automatic purity
+    icon: "mdi:scent"
+    category: config
+    dps:
+      - id: 107
+        type: boolean
+        name: switch
+  - entity: button
+    name: Change cat litter

```

Comment:
Is there not a duration sensor to go with this?

Reply:
So there is nothing keeping track of the time before you need to next change it? In most devices these types of buttons just reset a timer once you've manually emptied it, but I found a video for this device, and I see now it is actually performing the empting steps automatically, so I guess it makes sense without such a timer still.